From 1d1171da0caffbef9249b8f095f0b625a5a231ac Mon Sep 17 00:00:00 2001
From: Raphael Mounier <mounierr07@gmail.com>
Date: Tue, 20 Dec 2022 06:21:10 +0100
Subject: [PATCH 6/8] Fix a lot of Huawei EMUI9 denied under android 13

Fix tee denied (hisuite)
Add support off splash2 Huawei logging
Authorize a lot of process to read properties
Fix various denied
---
 sepolicy/huawei.te | 161 ++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 158 insertions(+), 3 deletions(-)

diff --git a/sepolicy/huawei.te b/sepolicy/huawei.te
index 2988ed6..d5c89a7 100644
--- a/sepolicy/huawei.te
+++ b/sepolicy/huawei.te
@@ -6,9 +6,9 @@ type dmd_device, file_type;
 allow hal_fingerprint_server dmd_device:chr_file rw_file_perms;
 allow hal_fingerprint_server sysfs:file rw_file_perms;
 allow tee hal_fingerprint_default:process { getattr };
-allow tee teecd_data_file:dir { search read write create getattr add_name open };
-allow tee teecd_data_file:file { read write create getattr open };
-allow tee system_data_file:dir { getattr };
+allow tee teecd_data_file:dir { rw_dir_perms create setattr };
+allow tee teecd_data_file:file { rw_file_perms create setattr };
+allow tee system_data_file:dir { rw_dir_perms create setattr };
 
 type oeminfo_nvm, domain;
 type oeminfo_nvm_device, file_type;
@@ -35,3 +35,158 @@ allowxperm vendor_init { teecd_data_file }:dir ioctl {
 type sys_block_mmcblk0, file_type;
 allow init sys_block_mmcblk0:lnk_file read;
 allow init sysfs_zram:lnk_file read;
+
+
+# --------------------- Not need to boot but this rules are denied  ----------------------------------------
+
+# Add support off splash2 Huawei logging
+type splash2_data_file, file_type;
+
+allow splash2_data_file self:filesystem { associate };
+allow init splash2_data_file:filesystem { mount relabelto relabelfrom };
+allow init splash2_data_file:dir { getattr mounton };
+allow vendor_init splash2_data_file:dir rw_dir_perms;
+allow vendor_init splash2_data_file:file rw_file_perms;
+allow vendor_init splash2_data_file:filesystem { getattr };
+allow installd splash2_data_file:filesystem { quotaget };
+allow rild splash2_data_file:dir rw_dir_perms;
+allow rild splash2_data_file:file rw_file_perms;
+allow kernel splash2_data_file:dir { rw_dir_perms create setattr };
+allow kernel splash2_data_file:file rw_file_perms;
+
+
+# Fix CPU gouvernor (rules are in xml file)
+type hwemerffu_service, domain;
+allow hwemerffu_service proc:file r_file_perms;
+
+
+# Fix all SEdenied
+# allow vold modem_secure_file:filesystem { unmount };
+
+type hinetmanager, fs_type, sysfs_type;
+allow netutils_wrapper hinetmanager:fd { use };
+allow netutils_wrapper hinetmanager:fifo_file { write };
+allow hinetmanager self:capability { dac_override };
+
+allow mediaprovider_app cache_file:dir r_dir_perms;
+
+
+type hal_usb_default, domain;
+allow hal_usb_default self:capability { dac_override };
+
+type sysfs_led, domain;
+allow init sysfs_led:file { setattr };
+
+
+#allow crash_dump bluetooth_data_file:file r_dir_perms;
+#allow crash_dump resourcecache_data_file:file { read open };
+
+allow vold sys_block_mmcblk0:file { write };
+allow vold sysfs_zram:file { write };
+
+allow kernel device:dir rw_dir_perms;
+allow kernel self:capability { mknod fsetid };
+allow kernel device:chr_file { create setattr };
+
+type teecd_data_file_system, data_file_type;
+allow vendor_init device:chr_file { getattr };
+allow vendor_init teecd_data_file_system:dir { rw_dir_perms create setattr relabelto };
+allow vendor_init system_data_file:dir { rw_dir_perms create setattr };
+allow vendor_init system_data_file:lnk_file { create };
+
+allow init teecd_data_file:dir { mounton };
+allow init teecd_data_file:filesystem { mount relabelto relabelfrom };
+allow init sys_block_mmcblk0:file { open write };
+
+allow fsck block_device:blk_file { read open write ioctl };
+allow teecd_data_file self:filesystem { associate };
+allow rootfs labeledfs:filesystem { associate };
+allow tee self:capability { sys_ptrace };
+
+allow kernel device:chr_file { open write };
+allow oeminfo_nvm block_device:blk_file { ioctl getattr open read write };
+allow gpsdaemon system_data_file:lnk_file { read };
+allow oeminfo_nvm device:chr_file { ioctl open read write };
+allow rild system_data_file:lnk_file { read };
+allow kernel system_data_root_file:dir r_dir_perms;
+allow kernel system_data_root_file:file rw_file_perms;
+allow kernel unlabeled:file { read open };
+
+
+allow hi110x_daemon rootfs:dir { open read };
+allow hi110x_daemon self:fifo_file { ioctl };
+
+
+allow hal_camera_default system_data_file:lnk_file { read };
+allow surfaceflinger bootanim:dir { search };
+allow surfaceflinger bootanim:file r_file_perms;
+allow hal_audio_default audioserver:fifo_file { write };
+allow installd self:capability { sys_ptrace };
+allow installd system_server:file { open read };
+allow installd system_server:dir { search };
+
+allow installd teecd_data_file:filesystem { quotaget };
+allow system_server sysfs_zram:lnk_file { read };
+allow system_server vendor_file:file r_file_perms;
+allow kernel sysfs_devices_system_cpu:file rw_file_perms;
+allow network_stack fs_bpf:file rw_file_perms;
+
+# OTPERASE = 0x4d19
+allowxperm oeminfo_nvm device:blk_file ioctl {
+  0x4d19
+};
+
+# TIOCGWINSZ = 0x5413
+allowxperm hi110x_daemon self:fifo_file ioctl {
+  TIOCGWINSZ
+};
+
+# BLKROSET = 0x125d 0x127c
+allowxperm phhsu_daemon device:blk_file ioctl {
+  BLKROSET
+  BLKDISCARDZEROES
+};
+
+
+allow shell self:netlink_kobject_uevent_socket { create setopt getopt bind read };
+allow init self:netlink_kobject_uevent_socket { create setopt getopt bind read };
+
+
+# Authorize a lot of process to read default, config and system prop
+type hal_health_default, domain;
+type hal_camera_default, domain;
+type hal_audio_default, domain;
+type hal_keymaster_default, domain;
+type hal_drm_widevine, domain;
+
+get_prop(rild, config_prop)
+get_prop(rild, system_prop)
+get_prop(hal_health_default, default_prop)
+get_prop(hal_health_default, config_prop)
+get_prop(hal_fingerprint_default, default_prop)
+get_prop(hal_camera_default, default_prop)
+get_prop(hal_camera_default, config_prop)
+get_prop(hal_audio_default, default_prop)
+get_prop(hal_audio_default, config_prop)
+get_prop(hal_audio_default, system_prop)
+get_prop(hal_keymaster_default, config_prop)
+get_prop(hal_drm_widevine, default_prop)
+
+type hi110x_daemon, domain;
+type wpa_hisi, domain;
+type aptouch_daemon, domain;
+type gpsdaemon, domain;
+type atcmdserver, domain;
+
+type hisecd, domain;
+type storage_info, domain;
+
+get_prop(hi110x_daemon, default_prop)
+get_prop(wpa_hisi, default_prop)
+get_prop(aptouch_daemon, system_prop)
+get_prop(gpsdaemon, default_prop)
+get_prop(oeminfo_nvm, default_prop)
+get_prop(atcmdserver, default_prop)
+get_prop(hisecd, default_prop)
+get_prop(mediaserver, vendor_default_prop)
+get_prop(storage_info, default_prop)
-- 
2.25.1

